From 0d6f66db3ee7dd1268fc4936d268fb747656832a Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Mon, 11 Apr 2016 16:55:02 +0200
Subject: [PATCH 2359/2367] tty: serial: 8250: don't take the trylock during
 oops

An oops with irqs off (panic() from irqsafe hrtimer like the watchdog
timer) will lead to a lockdep warning on each invocation and as such
never completes.
Therefore we skip the trylock in the oops case.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 drivers/tty/serial/8250/8250_port.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/8250/8250_port.c b/drivers/tty/serial/8250/8250_port.c
index ddf82d7..72e1e6af 100644
--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -3190,7 +3190,9 @@ void serial8250_console_write(struct uart_8250_port *up, const char *s,
 
 	touch_nmi_watchdog();
 
-	if (port->sysrq)
+	serial8250_rpm_get(up);
+
+	if (port->sysrq || oops_in_progress)
 		locked = 0;
 	else if (in_kdb_printk())
 		locked = spin_trylock_irqsave(&port->lock, flags);
-- 
2.7.4

